Public/Src/App/Bxl/BuildXLApp.cs

```diff
@@ -1229,15 +1236,13 @@ private static WarningManager CreateWarningManager(IWarningHandling configuratio
 
             public TrackingEventListener TrackingEventListener { get; private set; }
 
-            private string rootLogDirectory = null;
-
             public void ConfigureLogging(LoggingContext loggingContext)
             {
                 lock (m_lock)
                 {
                     Contract.Assume(!m_disposed);
 
-                    rootLogDirectory = m_configuration.LogsDirectory.IsValid ? m_configuration.LogsDirectory.ToString(m_pathTable) : null;
+                    RootLogDirectory = m_configuration.LogsDirectory.IsValid ? m_configuration.LogsDirectory.ToString(m_pathTable) : null;

```

Comment:
>RootLogDirectory = m_configuration.LogsDirectory.IsValid ? m_configuration.LogsDirectory.ToString(m_pathTable) : null; [](start = 20, length = 118)

nit: should this be happening in the constructor?  We are initializing `LogPath` there, but `RootLogDirectory` here...

Reply:
Good call. I didn't look particularly closely at it but it seems rootLogDirectory was previously dead code before I added this new reference on it. I moved to the constructor and also changed it to be based on the parent of the log file instead of the configuration which may or may not actually be set

---
In reply to: [316291646](https://github.com/microsoft/BuildXL/pull/754#discussion_r316291646) [](ancestors = 316291646)